zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobActivationBehavior.java

```diff
@@ -150,4 +158,35 @@ private void cloneJob(final JobRecord jobRecord, final JobRecord jobRecordClone)
     jobRecord.write(jobCopyBuffer, 0);
     jobRecordClone.wrap(jobCopyBuffer, 0, jobRecord.getLength());
   }
+
+  private boolean isAuthorized(
+      final JobActivationProperties jobActivationProperties, final JobRecord jobRecord) {
+
+    final var ownerTenantId = jobRecord.getTenantId();
+    final var tenantIds = jobActivationProperties.tenantIds().stream().toList();
+    if (!tenantIds.contains(ownerTenantId)) {
+      // don't push jobs to workers that don't request them from the job's tenant
+      return false;
+    }
+
+    final var authorizations = jobActivationProperties.authorizations();
+    final var authorizedTenantIds =
+        authorizationCheckBehavior.getAuthorizedTenantIds(authorizations);
+    final var authorizedProcessIds =
+        authorizationCheckBehavior.getAllAuthorizedScopes(
+            new AuthorizationRequest(
+                authorizations,
+                AuthorizationResourceType.PROCESS_DEFINITION,
+                PermissionType.UPDATE_PROCESS_INSTANCE,
+                ownerTenantId));
+
+    return authorizedTenantIds.isAuthorizedForTenantIds(tenantIds)
```

Comment:
🔧 It seems to me like this is too eager: The job is only owned by one tenant, it's enough if the stream is authorized for that specific tenant. If it also requests jobs for another tenant that it isn't authorized for, it can still receive jobs from tenants that it _is_ authorized for.

Reply:
I see, good point. Since this is a pretty much unavoidable change in behavior by now, I'd just do the sensible thing and push jobs that the client is actually authorized, even when it also requests jobs for an unauthorized tenant.

I think the other change in behavior is that previously, this check was only performed on stream creation and wouldn't update during the lifetime of the stream? Now it is always reacting to changes to the runtime authorization data. 

> Either way, I think it would be good to write a WARN log so users can see that they are requesting jobs for a tenant they aren't authorized for

I agree, let's just make sure it's not logged on every pushed job but ideally only the first time.